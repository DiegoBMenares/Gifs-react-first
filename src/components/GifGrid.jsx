import { GifItem } from "./GifItem";
import { useFetchGifs } from "../hooks/useFetchGifs";


export const GifGrid = ({category}) => {

    const {images,isLoading} = useFetchGifs(category);
    console.log('hola mundo 1');
    
    return (
        <>
            <h3>{ category }</h3>
            console.log('hola ');
            
            {
                isLoading && ( <h2>Cargando...</h2> )
            }
            
            <div className="card-grid">
                {
                    images.map( (images) => (
                        <GifItem 
                            key={images.id}
                            {...images}    
                            
                        />
                    ))
                }
            </div>
        </>
    )
}
