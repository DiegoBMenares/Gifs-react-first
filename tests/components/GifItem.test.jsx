/**
 * @jest-environment jsdom
 */

import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";


describe('Test de GifItem', () => { 

    const title = 'Goku';
    const url   = 'https://one-punch.com/saitama.jpg';


    test('Debe hacer match con el sanpshot', () => {
        
        const{container} = render(<GifItem title ={ title } url ={ url } />);
        expect (container).toMatchSnapshot();
    });

    test('Debe de mostrar la imagen con el URL y el ALT indicado', () => { 
    
        render(<GifItem title ={ title } url ={ url } />);
        //expect(screen.getByRole('img').alt ).toBe( title );
        
        const{src,alt} = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
    });

    test('Debe mostrar el titulo en el componete', () => { 

        render(<GifItem title ={ title } url ={ url } />);

        expect(screen.getByText(title)).toBeTruthy();
    })

});