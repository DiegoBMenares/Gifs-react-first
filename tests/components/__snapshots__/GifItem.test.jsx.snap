// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test de GifItem Debe hacer match con el sanpshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Goku"
      src="https://one-punch.com/saitama.jpg"
    />
    <p>
       
      Goku
       
    </p>
  </div>
</div>
`;
